packages/suite/src/components/suite/DeviceDisplay.tsx

```diff
@@ -0,0 +1,230 @@
+`;
+
+const StyledChunkedNextIcon = styled(Icon)<{ isPixelType: boolean }>`
+    position: relative;
+    bottom: ${props => (props.isPixelType ? '13' : '20')}px;
+    right: -67px;
+`;
+
+const StyledChunkedContinuesIcon = styled(Icon)<{ isPixelType: boolean }>`
+    position: relative;
+    top: ${props => (props.isPixelType ? '10' : '25')}px;
+    right: 68px;
+`;
+
+export interface DeviceDisplayProps {
+    address: string;
+    network?: string;
+}
+
+export const DeviceDisplay = ({ address, network }: DeviceDisplayProps) => {
+    const selectedDeviceInternalModel = useSelector(selectDeviceInternalModel);
+    const unavailableCapabilities = useSelector(selectDeviceUnavailableCapabilities);
+    const addressDisplay = useSelector(selectAddressDisplay);
+
+    const areChunksUsed =
+        addressDisplay === AddressDisplayOptions.CHUNKED && !unavailableCapabilities?.chunkify;
+    const isPixelType = selectedDeviceInternalModel !== DeviceModelInternal.T2T1;
+    const isPaginated = network === 'cardano';
+
+    const showChunksInRows = (chunks: string[][] | undefined, isNextAddress?: boolean) =>
+        chunks?.map((row, rowIndex) => (
+            <Row
+                // eslint-disable-next-line react/no-array-index-key
+                key={rowIndex}
+                isPixelType={isPixelType}
+                isAlignedRight={rowIndex === 0 && isNextAddress}
+            >
+                {row.map((chunk, chunkIndex) => (
+                    // eslint-disable-next-line react/no-array-index-key
+                    <Text isPixelType={isPixelType} key={chunkIndex}>
+                        {chunk}
+                    </Text>
+                ))}
+            </Row>
+        ));
+
+    const showChunks = (address: string) => {
+        const chunks = address.match(/.{1,4}/g);
+
+        const groupedChunks = chunks?.reduce((result, chunk, index) => {
+            const rowIndex = Math.floor(index / 4);
+
+            if (!result[rowIndex]) {
+                result[rowIndex] = [];
+            }
+
+            if (isPaginated && rowIndex === 3 && result[rowIndex].length === 3) {
+                result[rowIndex].push('');
+            } else if (isPaginated && rowIndex === 4 && result[rowIndex].length === 0) {
+                result[rowIndex].push('');
+            } else {
+                result[rowIndex].push(chunk);
+            }
+
+            return result;
+        }, [] as string[][]);
+
+        if (isPaginated) {
+            return (
+                <Content>
+                    <Chunks isPixelType={isPixelType}>
+                        {showChunksInRows(groupedChunks?.slice(0, 4))}
+                    </Chunks>
+                    <StyledChunkedNextIcon
+                        isPixelType={isPixelType}
+                        size={isPixelType ? 10 : 20}
+                        icon={isPixelType ? 'ADDRESS_PIXEL_NEXT' : 'ADDRESS_NEXT'}
+                        color={isPixelType ? '#ffffff' : '#959596'}
```

Comment:
nit: better use a theme color instead of `#959596` here

Reply:
It will change based on user's theme and I need this color to not change

Reply:
true!